chore(docs): engine-specific endpoints for public api (#130)
camunda/camunda-docs-cawemo: content/reference/rest-api/engine-diagrams/create.md

---
title: 'Engine Diagrams: create'
weight: 10

menu:
  main:
    name: 'create'
    identifier: 'rest-api-engine-diagrams-create'
    parent: 'rest-api-engine-diagrams'
    pre: 'Creates or updates an engine diagram.'
---

Creates a new engine diagram if it doesn't exist yet or updates an engine diagram with a new version.

# Request

## HTTP Request

```

> PUT https://cawemo.com/api/v1/engine-diagrams

Reviewer: Isn't here the project path parameter missing or to which project would the diagram be added?

Author: Good catch, it's missing.

```

## Request Body

In the request body, supply an [Engine Diagrams resource]({{< ref "/reference/rest-api/engine-diagrams/_index.md#resource-representation" >}}) with the following properties:

<table class="table table-striped">
  <tr>
    <th>Property Name</th>
    <th>Type</th>
    <th>Description</th>
    <th>Notes</th>
  </tr>
  <tr>
    <td>content</td>
    <td>string</td>
    <td>The engine diagram's BPMN 2.0 XML definition.</td>
    <td></td>
  </tr>
  <tr>
    <td>created (optional)</td>
    <td>date (UNIX Epoch time in milliseconds)</td>
    <td>The creation date assigned to the engine diagram's version.</td>
    <td>Has to lie in the past. Defaults to <i>now</i>.</td>
  </tr>
  <tr>
    <td>version</td>
    <td>number</td>

>     <td>The version of the engine diagram.</td>

Reviewer: 'The version of the process definition.' would be more aligned with the platform docs: https://docs.camunda.org/manual/7.15/reference/rest/process-definition/get/#result

Reviewer 2: If we want to use `process definition`, then we should replace `engine diagram` with this term everywhere.

Author: As agreed on in today's meeting we'll go with `process definition` instead of `engine diagram` everywhere.

---
title: 'Engine Diagrams: create'
weight: 10

menu:
  main:
    name: 'create'
    identifier: 'rest-api-engine-diagrams-create'
    parent: 'rest-api-engine-diagrams'
    pre: 'Creates or updates an engine diagram.'
---

Creates a new engine diagram if it doesn't exist yet or updates an engine diagram with a new version.

# Request

## HTTP Request

```

> PUT https://cawemo.com/api/v1/engine-projects/:project-name/engine-diagrams

Reviewer 2: Strictly speaking in terms of the rest API and maybe with the risk of missing some context here, I would like to ask why would we use `:project_name` since this should be a unique identifier?

To explain better where my question is coming from: Cawemo today does not enforce unique project/folder/file names in an organization/project or folder/project or folder.

Author: In today's meeting we agreed on using the project's id instead of the name as path variable.

```

## Path Parameters

<table class="table table-striped">
  <tr>
    <th>Name</th>
    <th>Type</th>
    <th>Description</th>
  </tr>
  <tr>
    <td>project-name</td>
    <td>string</td>

>     <td>Name of the engine project the diagram is located in.</td>

Reviewer 2: May sound a bit better, but feel free to dismiss the suggestion if you don't like it.
```suggestion
    <td>Name of the engine project that contains the diagram.</td>
```

Reviewer 2: And if this works, there is at least one more place that uses the same phrasing.
